core/src/validator.rs

```diff
@@ -1021,7 +1039,7 @@ impl Validator {
                 config.send_transaction_service_config.clone(),
                 max_slots.clone(),
                 leader_schedule_cache.clone(),
-                connection_cache.clone(),
+                tpu_connection_cache.clone(),
```

Comment:
So for staked nodes (which generally don't run rpc), this `tpu_connection_cache` is effectively unused, right?
(and I guess for rpc nodes (which generally are not staked), the other `fwd` cache is not used)

Wouldn't it be better to just have one single connection cache, and instead adjust the `pool` size based on if the node is running rpc or not?

Reply:
I would not rather rule that out -- a staked node can server RPC if they choose to do so.